admin/c2cgeoportal_admin/widgets.py

```diff
@@ -24,9 +24,23 @@ def serialize(self, field, cstruct, **kw):
         return super().serialize(field, cstruct, **kw)
 
 
+class ThemeOrderWidget(MappingWidget):
+
+    template = 'child'
+
+    def serialize(self, field, cstruct, **kw):
+        from c2cgeoportal_commons.models.main import TreeItem
+        if cstruct['id'] == colander.null:
```

Comment:
I don't the understand the purpose of this indirection.

Reply:
This is for the prototype (template for new item on client side), in such case we need a new object. Note that in theme we do not add new items, but deform need one prototype to render.